### PR TITLE
Bootstrap node receiving messages, satellite node sending messages

### PR DESCRIPTION
**Pull request by RokasGit**

### CODE REVIEW

File: src/logger.cpp [hunk 1]
```diff
@@ -27,4 +30,5 @@ void Logger::log(LogLevel level, const std::string &message)
     }
 
     logFile << Logger::getCurrentTime() << " " << levelStr << message << std::endl;
+    std::cout << Logger::getCurrentTime() << " " << levelStr << message << std::endl;
```
Comment (vdvivek) [inline src/logger.cpp]: Unnecessary

File: src/peer.cpp
Comment (vdvivek) [inline src/peer.cpp:1]: Use curved braces instead of brackets for initialization

File: src/peer_bootstrap_node.cpp [hunk 2]
```diff
@@ -25,40 +50,75 @@ void PeerBootstrapNode::sendData(const std::vector<uint8_t> &data)
...
+                                        std::to_string(remote_endpoint_.port()) +
+                                        " Size: " + std::to_string(bytes_transferred));
+
+        // Process the data (placeholder)
+        Logger::log(LogLevel::DEBUG, "Data: " + std::string(data.begin(), data.end()));
 
+        // Continue listening
+        socket_.async_receive_from(
+            asio::buffer(receive_buffer_), remote_endpoint_,
+            [this](const asio::error_code &error, std::size_t bytes_transferred)
```
Comment (vdvivek) [inline src/peer_bootstrap_node.cpp:65]: Why use a lambda around the function instead of the actual function

File: src/peer_satellite_node.cpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        satelliteNode.startListening();
+
+        bool running = true;
+        while (running)
+        {
+            // Send dummy data periodically
+            std::vector<uint8_t> dummy_data = {0x48, 0x65, 0x6C, 0x6C, 0x6F}; // "Hello"
+            satelliteNode.sendData(dummy_data);
+
+            std::this_thread::sleep_for(std::chrono::seconds(2)); // Avoid spamming
```
Comment (Prakash3008) [inline src/peer_satellite_node.cpp:115]: remove comments :D

File: src/peer_satellite_node.cpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        satelliteNode.setHostAddress(host_ip, host_port);
+        satelliteNode.setBootstrapAddress(bootstrap_ip, bootstrap_port);
+        satelliteNode.connect();
+        satelliteNode.startListening();
+
+        bool running = true;
+        while (running)
+        {
+            // Send dummy data periodically
+            std::vector<uint8_t> dummy_data = {0x48, 0x65, 0x6C, 0x6C, 0x6F}; // "Hello"
```
Comment (Prakash3008) [inline src/peer_satellite_node.cpp:112]: here too :D

File: src/peer_bootstrap_node.cpp [hunk 5]
```diff
@@ -11,12 +11,37 @@ void PeerBootstrapNode::connect()
 {
     Logger::log(LogLevel::INFO, "Connecting bootstrap node on " + host_ip_ + ":" + std::to_string(host_port_));
     // Connection code here
```
Comment (Prakash3008) [inline src/peer_bootstrap_node.cpp]: here too :D

File: src/peer_bootstrap_node.cpp [hunk 6]
```diff
@@ -25,40 +50,75 @@ void PeerBootstrapNode::sendData(const std::vector<uint8_t> &data)
...
-    bootstrapNode.connect();
-
-    bool running = true;
-    while (running)
+    try
     {
-        std::vector<uint8_t> data;
-        bootstrapNode.receiveData(data);
+        bootstrapNode.setHostAddress(host_ip, host_port);
+        bootstrapNode.connect();
```
Comment (vdvivek) [inline src/peer_bootstrap_node.cpp:108]: Still don't like this name

**Review by vdvivek — CHANGES_REQUESTED** [review]

**Review by Prakash3008 — DISMISSED** [review]
This is garbage x Niels

**Review by architpanigrahi — APPROVED** [review]
Approved.

File: src/logger.cpp (hunk 1 above)
Comment (RokasGit) [reply src/logger.cpp]: forgot to remove it, will do it, had to use it as when I had the file open it was destroying my pc q.q

File: src/peer_bootstrap_node.cpp (hunk 2 above)
Comment (RokasGit) [reply src/peer_bootstrap_node.cpp:65]: it's kinda of a standard for this part.

File: src/peer_bootstrap_node.cpp (hunk 6 above)
Comment (RokasGit) [reply src/peer_bootstrap_node.cpp:108]: It's how it is called theoretically, I also would probably prefer discovery node or something, but it is what it is.

File: src/peer_satellite_node.cpp (hunk 3 above)
Comment (RokasGit) [reply src/peer_satellite_node.cpp:115]: will need to do it this is actually dummy data, to test the workability of sending, so in the future more, as rn this is needed to understand what's going on. (0x48, 0x65, 0x6C, 0x6C, 0x6F) is not rly self explanatory if u know what I mean :D

File: src/peer_satellite_node.cpp (hunk 4 above)
Comment (RokasGit) [reply src/peer_satellite_node.cpp:112]: yea part of that, just a reminder will be gone in the future.

File: src/peer_bootstrap_node.cpp (hunk 5 above)
Comment (RokasGit) [reply src/peer_bootstrap_node.cpp]: should do that as I implemented that part XD

File: src/peer.cpp
Comment (RokasGit) [reply src/peer.cpp:1]: it's kind of a standard way to initialize this way.

**Review by ZeYanLiow — APPROVED** [review]

**Review by ZeYanLiow — APPROVED** [review]

**Pull request merged**